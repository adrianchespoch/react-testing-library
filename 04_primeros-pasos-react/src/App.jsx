import { useState } from 'react';
import PropTypes from 'prop-types';

import './App.css';
import './assets/tnrbjwasofaag8cm7l9v.jpg';
import './assets/react.svg';

function App({ title, initCounter }) {
  const [counter, setCounter] = useState(initCounter);

  // console.log('Render TODO el FC');  // Cuando CAMBIA el STATE se vuelve a Ejecutar el FC

  // // Si la f(x) esta fuera del FC no se reasisgna espacio en memoria con c/rerender
  const handleCounter = () => {
    // setCounter(counter + 1);
    setCounter(c => c + 1);
  };
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(initCounter);

  return (
    <div className="App">
      <h1 data-testid="test-title">{title}</h1>
      <h2>{counter}</h2>

      {/* <Counter /> */}
      <div className="card">
        <button className="btn-increase" onClick={handleCounter}>
          +1
        </button>
        <button id="btn-decrease" onClick={handleSubtract}>
          -1
        </button>
        <button aria-label="btn-reset" onClick={handleReset}>
          Reset
        </button>
      </div>

      <p>
        Edit <code>src/App.jsx</code> and save to test HMR
      </p>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </div>
  );
}

export default App;

// // // PropTypes
App.propTypes = {
  title: PropTypes.string.isRequired,
  initCounter: PropTypes.number.isRequired,
};

// // DefaultProps: Entran antes q los PropTypes, evitan error si es  isRequired
App.defaultProps = {
  title: 'No title!',
};
