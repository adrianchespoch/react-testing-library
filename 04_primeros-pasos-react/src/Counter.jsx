import { useState } from 'react';

export const Counter = () => {
  const [counter, setCounter] = useState(0);

  return (
    <div className="card">
      <button onClick={() => setCounter(counter + 1)}>
        count is: {counter}
      </button>
    </div>
  );
};
