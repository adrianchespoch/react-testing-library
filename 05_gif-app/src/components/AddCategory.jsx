import PropTypes from 'prop-types';

import { useForm } from '../hooks/useForm';

export const AddCategory = ({ onNewCategory }) => {
  const [formValues, handleInputChange, reset] = useForm({ toSearch: '' });
  const { toSearch } = formValues;

  const handleSubmit = e => {
    e.preventDefault();
    if (!toSearch.trim()) return;

    onNewCategory(toSearch.trim());
    reset();
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <input
        autoFocus
        type="text"
        name="toSearch"
        placeholder="Buscar gifs"
        value={toSearch}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
