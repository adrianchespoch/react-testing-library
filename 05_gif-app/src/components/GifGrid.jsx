import PropTypes from 'prop-types';

import { GifItem } from './GifItem';
import { useFetch } from '../hooks/useFetch';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetch(category);

  return (
    <>
      <h3>{category}</h3>

      <div className="card-grid">
        {isLoading && <p>Loading...</p>}

        {!isLoading &&
          images.map(image => <GifItem key={image.id} {...image} />)}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
