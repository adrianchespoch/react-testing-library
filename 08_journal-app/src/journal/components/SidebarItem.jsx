import { useMemo } from 'react';
import { useDispatch } from 'react-redux';
import { TurnedInNot } from '@mui/icons-material';
import {
  Grid,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
} from '@mui/material';
import { setActiveNote } from '../../store/journal';

export const SidebarItem = ({ title = '', body, id, date, imageUrls = [] }) => {
  const dispatch = useDispatch();

  const newTitle = useMemo(() => {
    return title.length > 17 ? title.substring(0, 18) + '...' : title;
  }, [title]);

  const handleSelectNote = () => {
    dispatch(setActiveNote({ id, title, body, date, imageUrls }));
  };

  return (
    <ListItem disablePadding onClick={handleSelectNote}>
      <ListItemButton>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>

        <Grid container>
          <ListItemText primary={newTitle}></ListItemText>
          <ListItemText primary={body}></ListItemText>
        </Grid>
      </ListItemButton>
    </ListItem>
  );
};
