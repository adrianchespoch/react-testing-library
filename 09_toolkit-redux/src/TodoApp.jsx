import { useState } from 'react';
import {
  useGetTodoQuery,
  // useGetTodosQuery,
} from './store/slices/apis/todosApi';

export const TodoApp = () => {
  const [todoId, setTodoId] = useState(1);

  // const { data: todos = [], isLoading } = useGetTodosQuery();
  const { data: todo, isLoading } = useGetTodoQuery(todoId);

  const prevTodo = () => {
    if (todoId <= 1) return;
    setTodoId(todoId - 1);
  };

  return (
    <>
      <h1>Todos - RTK Query</h1>
      <hr />

      <h4>isLoading: {isLoading ? 'True' : 'False'}</h4>

      <pre>{JSON.stringify(todo, null, 3)}</pre>

      <button onClick={prevTodo}>Prev todo</button>
      <button onClick={() => setTodoId(todoId + 1)}>Next todo</button>

      {/* <ul>
        {todos.map(todo => (
          <li key={todo.id}>
            <strong>{todo.completed ? 'DONE' : 'Pending'}: </strong>
            {todo.title}
          </li>
        ))}
      </ul> */}
    </>
  );
};
