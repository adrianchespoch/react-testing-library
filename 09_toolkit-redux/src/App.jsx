import { useDispatch, useSelector } from 'react-redux';

import { decrement, increment, incrementByAmount } from './store/slices/counter';
import './App.css';

function App() {
  // Extraemos lo q definimos en el initState del slice
  const { counter } = useSelector(state => state.counter);

  // ya sabe a q reducer va a llegar
  const dispatch = useDispatch();

  return (
    <div className="App">
      <h1>Count is: {counter}</h1>

      <div className="card">
        <button onClick={() => dispatch(increment())}>Increment</button>
        <button onClick={() => dispatch(decrement())}>Decrement</button>
        <button onClick={() => dispatch(incrementByAmount(3))}>Increment by 2</button>
      </div>
    </div>
  );
}

export default App;
