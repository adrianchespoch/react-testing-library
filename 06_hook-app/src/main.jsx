import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './01_useState/CounterApp';
import { CounterWithCustomHook } from './01_useState/CounterWithCustomHook';
import { FormWithCustomHook } from './02_useEffect/FormWithCustomHook';
import { SimpleForm } from './02_useEffect/SimpleForm';
import { MultipleCustomHook } from './03-examples/MultipleCustomHook';
import { FocusScreen } from './04_useRef/FocusScreen';
import { Layout } from './05_useLayoutEffect/Layout';
import { CallbackHook } from './06-memos/CallbackHook';
import { MemoHook } from './06-memos/MemoHook';
import { Memorize } from './06-memos/Memorize';
import { Padre } from './07-memo-task/Padre';

import './index.css';
// import './08_useReducer/intro-reducer';
import { TodoApp } from './08_useReducer/TodoApp';
import { MainApp } from './09_useContext/MainApp';
import { AppRouter } from './09_useContext/AppRouter';
import { UserProvider } from './09_useContext/context/UserProvider';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  // {/* <App /> */}
  // {/* <CounterApp /> */}
  //{/* <CounterWithCustomHook /> */}

  // // useEffect
  // <SimpleForm />
  // <FormWithCustomHook />

  // // Example
  // <MultipleCustomHook />

  // // useRef
  // <FocusScreen />

  // // useLayoutEffect
  // <Layout />

  // // memo()
  // <Memorize />

  // // useMemo
  // <MemoHook />

  // // useCallback
  // <CallbackHook />
  // <Padre />

  // // useReducer()
  // <TodoApp />

  // // useContext()
  // <MainApp />
  <AppRouter />

  // </React.StrictMode>
);
