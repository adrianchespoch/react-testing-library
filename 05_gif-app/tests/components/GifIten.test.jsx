import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/components';

describe('Prubeas sobre GifIten.jsx', () => {
  const title = 'One Punch';
  const url = 'https://giphy.com/';

  test('should hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test('should return the image with the specified URL and ALT', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();

    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should return the component title', () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
