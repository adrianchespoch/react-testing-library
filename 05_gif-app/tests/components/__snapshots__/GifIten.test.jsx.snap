// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubeas sobre GifIten.jsx should hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://giphy.com/"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
